libs/auth-api-lib/src/lib/delegations/admin/delegation-admin-custom.service.ts

```diff
@@ -1,99 +1,226 @@
+      fromReferenceId: fromReferenceId.value,
+      toReferenceId: toReferenceId.value,
+    }
+  }
+
   async getAllDelegationsByNationalId(
     nationalId: string,
   ): Promise<DelegationAdminCustomDto> {
-    const incomingDelegations = await this.delegationModel.findAll({
-      where: {
-        toNationalId: nationalId,
-      },
-      include: [
-        {
-          model: DelegationScope,
-          required: true,
-          include: [
-            {
-              model: ApiScope,
-              as: 'apiScope',
-              required: true,
-              where: {
-                enabled: true,
-              },
-              include: [
-                {
-                  model: ApiScopeDelegationType,
-                  required: true,
-                  where: {
-                    delegationType: AuthDelegationType.Custom,
-                  },
+    const incomingDelegationPromises = []
+
+    incomingDelegationPromises.push(
+      this.delegationsIncomingCustomService.findAllValidIncoming({
+        nationalId: nationalId,
+        validity: DelegationValidity.ALL,
+      }),
+    )
+
+    incomingDelegationPromises.push(
+      this.delegationsIncomingCustomService.findAllValidGeneralMandate({
+        nationalId: nationalId,
+      }),
+    )
+
+    const [
+      incomingDelegationSets,
+      outgoingDelegations,
+      generalOutgoingDelegations,
+    ] = await Promise.all([
+      await Promise.all(incomingDelegationPromises),
+      this.delegationModel.findAll({
+        where: {
+          fromNationalId: nationalId,
+        },
+        include: [
+          {
+            model: DelegationScope,
+            required: true,
+            include: [
+              {
+                model: ApiScope,
+                required: true,
+                as: 'apiScope',
+                where: {
+                  enabled: true,
                 },
-              ],
+                include: [
+                  {
+                    model: ApiScopeDelegationType,
+                    required: true,
+                    where: {
+                      delegationType: AuthDelegationType.Custom,
+                    },
+                  },
+                ],
+              },
+            ],
+          },
+        ],
+      }),
+      this.delegationModel.findAll({
+        where: {
+          fromNationalId: nationalId,
+        },
+        include: [
+          {
+            model: DelegationDelegationType,
+            required: true,
+            where: {
+              delegationTypeId: AuthDelegationType.GeneralMandate,
             },
-          ],
+          },
+        ],
+      }),
+    ])
+
+    return {
+      incoming: ([] as DelegationDTO[]).concat(...incomingDelegationSets),
+      outgoing: [
+        generalOutgoingDelegations.map((d) =>
+          d.toDTO(AuthDelegationType.GeneralMandate),
+        ),
+        outgoingDelegations.map((delegation) => delegation.toDTO()),
+      ].flat(),
```

Comment:
You could also spread

```suggestion
      outgoing: [
        ...generalOutgoingDelegations.map((d) =>
          d.toDTO(AuthDelegationType.GeneralMandate),
        ),
        ...outgoingDelegations.map((delegation) => delegation.toDTO()),
      ],
```